examples/dgfip_c/ml_primitif/Makefile

```diff
@@ -6,11 +6,118 @@ else
     TEST_FILES=$(TESTS_DIR)*
 endif
 
-prim:
-	./build.sh
+# DGFiP C backend works only with these specifications at the moment.
+MPP_FILE=../../../mpp_specs/dgfip_base.mpp
+MPP_FUNCTION=dgfip_calculation
 
-run_tests: prim
-	 ./prim ${TEST_FILES} || (echo "Test failure" && false)
+MLANG_DGFIP_C_OPTS=\
+	--mpp_file=$(MPP_FILE) \
+	--mpp_function=$(MPP_FUNCTION)
+
+# Options supplémentaires pour le backend Mlang/DGFiP
+# Note: les flags -Ailiad, -m, -g et -k sont déjà définis
+DGFIPFLAGS=-X,-O  # = environnement DGFiP
+
+MLANG=$(MLANG_BIN) $(MLANG_DEFAULT_OPTS) $(MLANG_DGFIP_C_OPTS)
+
+# Options pour le compilateur C
+# Attention, très long à compiler avec GCC en O2/O3
+# Le flag sur la profondeur des blocs ne semble plus nécessaire.
+COMMON_CFLAGS=-std=c89 -pedantic
+ifeq ($(C_COMPILER), clang)
+	COMPILER_SPECIFIC_CFLAGS=-O2 -fbracket-depth=2048
+else ifeq ($(C_COMPILER), gcc)
+    COMPILER_SPECIFIC_CFLAGS=-O1
+endif
+CFLAGS=$(COMMON_CFLAGS) $(COMPILER_SPECIFIC_CFLAGS)
+
+# Options pour le compilateur OCaml
+OCAMLFLAGS=
+#OCAMLFLAGS="-g -inline 0"
+# Pour instrumenter la couverture de code, il est nécessaire d'installer le paquet OCaml bisect_ppx
+# 1 pour activer la couverture de code
+WITH_BISECT?=0
+ifeq ($(WITH_BISECT), 1)
+    BISECT_PATH:=$(shell ocamlfind query bisect_ppx)
+    ifeq ($(BISECT_PATH),)
+      $(error $(BISECT_PATH) \
+	  Pour instrumenter la couverture de code, il est nécessaire d'installer le paquet OCaml bisect_ppx)
+    endif
+endif
+
+##################################################
+# Initialisation
+##################################################
+
+calc:
+	mkdir $@
+
+##################################################
+# C sources generation
+##################################################
+
+calc/var_static.c.inc: calc
+	@echo "Copie des fichiers C statiques"
+	cp static/* calc/
+
+# Ancien makefile ml_primitif
+# -----------------------------
+# Note: compilation avec -g et -k1 OBLIGATOIREMENT
+calc/enchain.c: $(SOURCE_FILES)
+	@echo "Compilation des fichiers M avec Mlang (YEAR=$(YEAR) MPP_FUNCTION=$(MPP_FUNCTION))"
+	@$(MLANG) \
+		--dgfip_options=-g,-k1,-b0,-Ailiad,-m$(YEAR),$(DGFIPFLAGS)\
```

Comment:
Really I don't know why the hell I reported a `-b0`which was not in the shell script. It passed the test using the "old" test method, but of course not the TGV comparison from PR#203